src/libsystemd-network/dhcp6-option.c

```diff
@@ -106,7 +106,7 @@ int dhcp6_option_append_ia(uint8_t **buf, size_t *buflen, const DHCP6IA *ia) {
                 return -EINVAL;
         }
 
-        if (*buflen < len)
+        if (*buflen < offsetof(DHCP6Option, data) + len)
```

Comment:
So it seems that it's not uncommon for `dhcp6-client` to get the length wrong. The receiving side is more or less covered by the dhcp6-client fuzzer (which has helped to find three similar issues this month). Maybe it would make sense to somehow cover the sending side too.